### ISSUE TITLE
Linking of unsatisfied local versions.

### ISSUE THREAD

**Issue by pleunv** [guigrpa/oao#34]
Given this scenario:
```
- packages/module-one@2.0.0
- packages/app-one
    - deps: module-one@1.0.0
- packages/app-two
    - deps: module-one@2.0.0
```

The default behavior seems to be to `link` `module-one@2.0.0` into both `app-one` and `app-two`, even though `app-one` is only satisfied by `1.0.0`. I'm a bit surprised this is happening (and that it also seems to be the case in lerna), as I would expect `oao` to try to fetch a version of `module-one` that satisfies its `package.json` instead once it notices that the local version does not. Right now you could end up with hard to track issues during development if one of the apps is using an incompatible version, especially since this also does not seem to be picked up by `oao outdated` (which seems to be because `yarn outdated` does not like symlinked modules). Is there any reasononing behind this behavior?

I guess this is also partly covered by #16 and the `yarn knit` discussion, but would there perhaps be a workaround that can be used in the meantime? Could `oao outdated` for example warn for version inconsistencies in consuming packages so that people running a publish are at least aware of an incompatibility?

**Comment by guigrpa** [comment]
I'm not sure it's useful to introduce the complexity of downloading a package in some cases and symlinking it in others, and also considering that Lerna doesn't do it either. In general, most use cases I'd envisage are happy with whatever is the latest version of the internal sub-packages for development.

What I can do is add warnings to `oao outdated` (as you suggest) and `oao bootstrap`, indicating that the dependency's version doesn't satisfy its user. Would that suit your use case?

Regarding `oao outdated`, it runs `yarn outdated` only after (temporarily) removing internal sub-packages (those that are symlinked) from the `package.json`. So it should work (in my computer, it doesn't, but due to [this bug](https://github.com/yarnpkg/yarn/issues/2871).

Finally, could you maybe share which monorepo you're using or testing `oao` on?

**Comment by guigrpa** [comment]
Update: yarn issue solved for me in v0.22.0

**Comment by guigrpa** [comment]
`oao outdated` warning just released in v0.8.1.
